pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1070,6 +1080,20 @@ func (og *operationGenerator) GenerateMapVolumeFunc(
 			return volumeToMount.GenerateError("MapVolume.MapDevice failed", mapErr)
 		}
 
+		// For CSI driver, devicePath should be checked here by checking symlink from
+		// {volumeMapPath}/{volName}.
```

Comment:
Why do we need to do this? Can we have `MapDevice` to return device path? For in-tree plugins it will simply return whatever was passed and for CSI it will return device path after publish.

Reply:
Fixed by directly using the `publishPath` as `devicePath`. (With ignoring error for resolving symlink.)
I thought `publishPath` also need to be resolved for containerized kubelet, so I put this logic here.
However, this assumption won't be true. (I think that `publishPath` could be handled by containerized kubelet.)

Only downside by doing this will be failure in handling error for resolving symlink for in-tree driver.